### PR TITLE
Server state machine

### PR DESCRIPTION
**Pull request by strikray**
Initial code for server state machine. Still needs work, but wanted to get up what I have.

### CODE REVIEW

File: server/game_type.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/python
+
+#CS544 - Computer Networks
+#Group 8
+#BGP Protocol
+#Team Members: Brian Quinn, TJ Rhodes, Ryan Mann, Marc Thomson
+
+#Module name: message
```
Comment (brianjquinn) [inline server/game_type.py:8]: copy and pasted i assume

File: server/state_machine.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/python
+
+#CS544 - Computer Networks
+#Group 8
+#BGP Protocol
+#Team Members: Brian Quinn, TJ Rhodes, Ryan Mann, Marc Thomson
+
+#Module name: message
```
Comment (brianjquinn) [inline server/state_machine.py:8]: copy and pasted I assume

File: server/state_machine.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+#Description: defines a server state machine object and its state transitions
+
```
Comment (brianjquinn) [inline server/state_machine.py:10]: also maybe include in this comment that there will be 1 per client

File: server/game_type.py (hunk 1 above)
Comment (strikray) [inline server/game_type.py:8]: yes

File: server/state_machine.py (hunk 2 above)
Comment (strikray) [inline server/state_machine.py:8]: yes

**Pull request merged**